src/components/chip/chip.tsx

```diff
@@ -0,0 +1,279 @@
+ * or clearly bundled into a group.
+ * :::
+ *
+ * @private
+ * @exampleComponent limel-example-chip-button
+ * @exampleComponent limel-example-chip-link
+ * @exampleComponent limel-example-chip-icon-colors
+ * @exampleComponent limel-example-chip-badge
+ * @exampleComponent limel-example-chip-filter
+ * @exampleComponent limel-example-chip-removable
+ * @exampleComponent limel-example-chip-aria-role
+ */
+@Component({
+    tag: 'limel-chip',
+    shadow: true,
+    styleUrl: 'chip.scss',
+})
+export class Chip {
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * ID of the chip. Must be unique.
+     */
+    @Prop()
+    id: number | string;
+
+    /**
+     * Label displayed on the chip
+     */
+    @Prop()
+    public text: string;
+
+    /**
+     * Icon of the chip.
+     */
+    @Prop()
+    public icon: string | Icon;
+
+    /**
+     * If supplied, the chip will become a clickable links.
+     */
+    @Prop()
+    link?: Omit<Link, 'text'>;
+
+    /**
+     * The value of the badge, displayed on the chip.
+     */
+    @Prop()
+    badge?: string | number;
+
+    /**
+     * Set to `true` to disable the chip.
+     */
+    @Prop({ reflect: true })
+    public disabled = false;
+
+    /**
+     * Set to `true` to render the chip as a static UI element.
+     * Useful when the parent component has a `readonly` state.
+     */
+    @Prop({ reflect: true })
+    public readonly = false;
+
+    /**
+     * Set to `true` to visualize the chip in a "selected" state.
+     * This is typically used when the chip is used in a chip-set
+     * along with other chips.
+     */
+    @Prop({ reflect: true })
+    public selected = false;
+
+    /**
+     * Set to `true` to render a remove button on the chip.
+     */
+    @Prop({ reflect: true })
+    public removable = false;
+
+    /**
+     * Set to `filter` to render the chip with a distinct style
+     * suitable for visualizing filters.
+     */
+    @Prop({ reflect: true })
+    public type: 'filter';
+
+    /**
+     * Fired when clicking on buttons (not links!)
+     * inside the breadcrumbs.
+     */
+    @Event()
+    public remove: EventEmitter<Chip>;
+
+    @Element()
+    private host: HTMLLimelChipElement;
+
+    private button: HTMLButtonElement;
+    private anchor: HTMLAnchorElement;
+
+    public componentWillLoad() {
+        makeEnterClickable(this.host as HTMLElement);
+    }
+
+    public disconnectedCallback() {
+        this.removeEnterClickable();
+    }
+
+    public render() {
+        if (this.link) {
+            return this.renderAsLink();
+        }
+
+        return this.renderAsButton();
+    }
+
+    private renderAsButton = () => {
+        return [
+            <button
+                class="chip"
+                role="button"
+                disabled={this.disabled || this.readonly}
```

Comment:
This doesn't make sense to me. When the component is readonly, there shouldn't be a remove-button rendered, but the component should still be clickable, right?

Reply:
I don't think we want it to be clickable. Think about a readonly chip-set. 🤔 Maybe I'm wrong. But throughout our components, we have used `readonly` to render some sort of "visualization-only" mode. More info here: https://github.com/Lundalogik/lime-elements/pull/2412#pullrequestreview-1541249268

Reply:
Hmm… oh… yeah… I guess it depends on what functionality is tied to clicking the chip… if it's a filter-chip, then no, it shouldn't be possible to click it, because it shouldn't be possible to change the chip's `selected` state. But if the chip represents a related object in a feed-item, it should still be possible to click on it to navigate to the related object, right?

Although, you perhaps don't consider that "readonly". That's just "the normal way chips are, when they don't have `removable=true`"? 🤔

Reply:
Exactly. In the feed, they are just fully functional and interactive chips. I'll resolve the similar comment further down, and you can resolve this when you have read it ;)

Reply:
yes, we need to control the readonly from outside, e.g. the consuming component like `limel-chip-set`. Also imagine a tag list of static hashtags, or static list of people joining an event. they could all be non-interactive set of chips